skfp/distances/tanimoto.py

```diff
@@ -0,0 +1,317 @@
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    Tanimoto similarity for binary vectors.
+
+    Computes the Tanimoto similarity [1]_ for binary data between two input arrays
+    or sparse matrices using the Jaccard index. The calculated similarity falls within
+    the explicit range [0, 1], with passing all-zero vectors to this function resulting
+    in a similarity of 1.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    ----------
+    similarity : float
+        Tanimoto similarity between vec_a and vec_b.
+
+    References
+    ----------
+    .. [1] `Bajusz, D., Rácz, A. & Héberger, K.
+       "Why is Tanimoto index an appropriate choice for fingerprint-based similarity calculations?"
+       J Cheminform, 7, 20 (2015).
+       <https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0069-3>`_
+
+    Examples
+    ----------
+    >>> from skfp.similarities import tanimoto_binary_similarity
+    >>> vec_a = np.array([1, 0, 1])
+    >>> vec_b = np.array([1, 0, 1])
+    >>> sim = tanimoto_binary_similarity(vec_a, vec_b)
+    >>> sim
+    1.0
+
+    >>> vec_a = csr_array([[1, 0, 1]])
+    >>> vec_b = csr_array([[1, 0, 1]])
+    >>> sim = tanimoto_binary_similarity(vec_a, vec_b)
+    >>> sim
+    1.0
+    """
+    _check_nan(vec_a)
+    _check_nan(vec_b)
+
+    if np.sum(vec_a) == 0 == np.sum(vec_b):
+        return 1.0
+
+    if isinstance(vec_a, csr_array) and isinstance(vec_b, csr_array):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+
+        intersection: float = vec_a_bool.multiply(vec_b_bool).sum()
+        union: float = vec_a_bool.sum() + vec_b_bool.sum() - intersection
+
+        return intersection / union
+
+    elif isinstance(vec_a, np.ndarray) and isinstance(vec_b, np.ndarray):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+
+        return 1 - jaccard(vec_a_bool, vec_b_bool)
+    else:
+        raise TypeError(
+            f"Both vec_a and vec_b must be of the same type: either numpy.ndarray "
+            f"or scipy.sparse.csr_array, got {type(vec_a)} and {type(vec_b)}"
+        )
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def tanimoto_binary_distance(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    Tanimoto distance for binary vectors.
+
+    Computes the Tanimoto distance for binary data between two input arrays or sparse matrices
+    by subtracting the similarity value from 1. The calculated distance falls within
+    the range [0, 1], with passing all-zero vectors to this function resulting in a distance of 0.
```

Comment:
End sentence after `[0, 1]` (BTW, it should be in backticks, so it renders nicely), and start next sentence as "Passing all-zero..."